pkg/printers/internalversion/printers.go

```diff
@@ -1154,11 +1154,26 @@ func formatHosts(rules []networking.IngressRule) string {
 	return ret
 }
 
-func formatPorts(tls []networking.IngressTLS) string {
-	if len(tls) != 0 {
-		return "80, 443"
+func formatPorts(obj *networking.Ingress) string {
+	allowHTTP := true
+	if annotationValue, exists := obj.Annotations["kubernetes.io/ingress.allow-http"]; exists {
```

Comment:
is this officially part of the ingress API?

Reply:
Good call, I guess someone from networking can weigh in.

The current code just hard codes the assumption that if you have TLS, ports are "443, 80" and otherwise just "80".

From what I can tell, `kubernetes.io/ingress.allow-http` is valid at least for ingress-gce:
https://github.com/kubernetes/ingress-gce/blob/c7c2a08366ed113009b60f88aca8760f43034c93/pkg/annotations/ingress.go#L30-L35.

Reply:
To be honest, and probably this is what you're getting at, it seems like the port numbers should not even be determined within the printer.

Seems like the ports should be reported out as status from the various ingress controllers, because in theory it would not even have to be 443 and 80.